deblur/deblurring.py

```diff
@@ -0,0 +1,167 @@
+    list of Sequence
+
+    Raises
+    ------
+    ValueError
+        If no sequences where found in `input_seqs`
+        If all the sequences do not have the same length either aligned or
+        unaligned.
+    """
+    seqs = [Sequence(label, seq) for label, seq in input_seqs]
+
+    if len(seqs) == 0:
+        raise ValueError("No sequences found!")
+
+    # Check that all the sequence lengths (aligned and unaligned are the same)
+    aligned_lengths = set(s.length for s in seqs)
+    unaligned_lengths = set(s.unaligned_length for s in seqs)
+
+    if len(aligned_lengths) != 1 or len(unaligned_lengths) != 1:
+        raise ValueError(
+            "Not all sequence have the same length. Aligned lengths: %s, "
+            "sequence lengths: %s"
+            % (", ".join(map(str, aligned_lengths)),
+               ", ".join(map(str, unaligned_lengths))))
+
+    seqs = sorted(seqs, key=attrgetter('frequency'), reverse=True)
+    return seqs
+
+
+def deblur(input_seqs, read_error=0.05, mean_error=None, error_dist=None,
+           indel_prob=0.01, indel_max=3):
+    """Deblur the reads
+
+    Parameters
+    ----------
+    input_seqs : iterable of (str, str)
+        The list of input sequences in (label, sequence) format
+    read_error : float, optional
+        The read error rate. Default: 0.05
+    mean_error : float, optional
+        The mean error, used for original sequence estimate. Default: same
+        value as `read_error`
+    error_dist : list of float, optional
+        A list of error probabilities. The length of the list determines the
+        amount of hamming distances taken into account. Default: None, computed
+        from mean error and sequence length.
+    indel_prob : float, optional
+        Indel probability (same for N indels). Default: 0.01
+    indel_max : int, optional
+        The maximal number of indels expected by errors. Default: 3
+
+    Results
+    -------
+    list of Sequence
+        The deblurred sequences
+
+    Notes
+    -----
+    mean_error is used only for normalizing the peak height before deblurring,
+    whereas read_error is used for calculating the expected number of errors
+    for each position.
+    The array 'error_dist' represents the error distribution, where
+    Xi = max frequency of error hamming. The length of this array - 1 limits
+    the hamming distance taken into account, i.e. if the length if `error_dist`
+    is 10, sequences up to 10 - 1 = 9 hamming distance will be taken into
+    account
+    """
+    # Get the sequences
+    seqs = get_sequences(input_seqs)
+
+    # If mean error is not provided, use the same value as read_error
+    mean_error = mean_error if mean_error else read_error
+
+    # if error_list not supplied, use the default (22 mock mixture setup)
+    mod_factor = pow((1 - mean_error), seqs[0].unaligned_length)
+    if not error_dist:
+        error_dist = np.array(
+            [1.0 / mod_factor, pow(read_error, 1) / mod_factor, 0.01, 0.01,
```

Comment:
Is the parenthesis in the right place? Otherwise, the second terms is `read_error**1 == read_error`, so the power is not necessary.